### ISSUE TITLE
[QUESTION] NVCV_ERROR_INVALID_ARGUMENT: Layout dimensions must match shape dimensions

### ISSUE THREAD

**Issue by junkojunko** [CVCUDA/CV-CUDA#68]
this is my code，how can i solve this question?

`import torch
import torchnvjpeg
device_id = 0
batch_size = 1
max_cpu_threads = 8
max_image_size = 8000*8000*3
decoder = torchnvjpeg.Decoder(
    0,
    0,
    True,
    device_id,
    batch_size,
    max_cpu_threads,
    max_image_size,
    torch.cuda.current_stream(device_id)
)
image_data = open('imgs/0210_2023_03_18_14_58_05.jpg','rb').read()
image_tensor = decoder.decode([image_data])
rw,rh = _scale_size((image_tensor.shape[1],image_tensor.shape[0]),0.5)
cvcuda_input_tensor = cvcuda.as_tensor(image_tensor,"NHWC")

# The Code below reports this error
cvcuda_resize_tensor = cvcuda.resize(
    cvcuda_input_tensor,
    (
        batch_size,
        rw,
        rh,
        3,
    ),
    cvcuda.Interp.LINEAR,
)
`

---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
/tmp/ipykernel_14987/4181054904.py in <module>
----> 1 cvcuda_resize_tensor = cvcuda.resize(
      2     cvcuda_input_tensor,
      3     (
      4         batch_size,
      5         rh,
      6         rw,

RuntimeError: NVCV_ERROR_INVALID_ARGUMENT: Layout dimensions must match shape dimensions

**Comment by lofyol** [comment]
请问你解决这个问题了吗

**Comment by jarun-title** [comment]
Try checking if your input tensor shape to cvcuda.as_tensor() is 4 dimensions. I got the same error when I use 'NHWC' with three dimensions tensor, it should be just 'HWC'.

**Comment by dsuthar-nvidia** [comment]
Yes, for 3 dimensional inputs it should be either HWC or CHW. For four dimensional inputs, i.e. with batch-size, it can be NCHW or NHWC. Here is a working example that can resize images using cvcuda:

```
import torch
import torchvision
import cvcuda

# Read image using PyTorch
image = torchvision.io.read_image("example.jpg")  # Returns a CHW tensor

# Move to GPU and convert to NHWC format for CV-CUDA
image_gpu = image.permute(1, 2, 0).unsqueeze(0).cuda()  # Convert to NHWC and add batch dimension

# Create CV-CUDA tensor from PyTorch tensor
cvcuda_tensor = cvcuda.as_tensor(image_gpu, "NHWC")

# Define target size (batch, height, width, channels)
target_size = (1, 224, 224, 3)

# Resize using CV-CUDA
resized_tensor = cvcuda.resize(
    cvcuda_tensor,
    target_size,
    cvcuda.Interp.LINEAR
)

# If you need to convert back to PyTorch tensor
resized_torch = torch.as_tensor(resized_tensor.cuda())

print(f"Original shape: {cvcuda_tensor.shape}")
print(f"Resized shape: {resized_tensor.shape}")
```
